source/Outfit.cpp

```diff
@@ -87,7 +99,20 @@ void Outfit::Load(const DataNode &node)
 				licenses.push_back(grand.Token(0));
 		}
 		else if(child.Size() >= 2)
-			attributes[child.Token(0)] = child.Value(1);
+		{
+			// These attributes are not allowed to have values <= -1. Catch any
+			// violations upon loading the game so that the corrections don't
+			// need to be repeatedly calculated later.
+			bool illegalAttribute = false;
+			if(PROTECTION_TYPES.count(child.Token(0)) != 0)
```

Comment:
Sorry, I meant you can just use the implicit boolean conversion of `set::count`.

Amending my other suggestion, something like
```c++
const string &key = child.Token(0);
attributes[key] = PROTECTION_TYPES.count(key) ? max(-.99, child.Value(1)) : child.Value(1);
```
would do the job.